### PR TITLE
test(connector): [Adyen] Add UI test for Adyen Payment methods

### PR DESCRIPTION
**Pull request by swangi-kumari**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added and fixed ui test for bank debits, bank redirects and wallets

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
adyen_ui_test

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by srujanchikke — APPROVED** [review]

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 1]
```diff
@@ -171,6 +209,261 @@ async fn should_make_adyen_alipay_hk_payment(c: WebDriver) -> Result<(), WebDriv
...
+async fn should_make_adyen_bancontact_card_payment(
+    web_driver: WebDriver,
+) -> Result<(), WebDriverError> {
+    let conn = AdyenSeleniumTest {};
+    conn.make_redirection_payment(
+        web_driver,
+        vec![
+            Event::Trigger(Trigger::Goto(&format!("{CHEKOUT_BASE_URL}/saved/68"))),
+            Event::Trigger(Trigger::Click(By::Id("card-submit-btn"))),
+            Event::Trigger(Trigger::SendKeys(By::Id("username"), "admin")),
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen_uk_ui.rs:324]: move username and password to connector_auth file

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 2]
```diff
@@ -171,6 +209,261 @@ async fn should_make_adyen_alipay_hk_payment(c: WebDriver) -> Result<(), WebDriv
...
+            Event::Trigger(Trigger::SendKeys(By::Id("password"), "Juspay@123")),
+            Event::Trigger(Trigger::Click(By::ClassName("button"))),
+            Event::Assert(Assert::IsPresent("failed")),
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen_uk_ui.rs:327]: are we expecting it to fail always?

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 3]
```diff
@@ -216,4 +521,89 @@ fn should_make_adyen_alipay_hk_payment_test() {
...
+#[ignore]
+#[test]
+#[serial]
+fn should_make_adyen_giropay_payment_test() {
+    tester!(should_make_adyen_giropay_payment);
+}
+
+#[test]
+#[serial]
+fn should_make_adyen_walley_payment_test() {
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen_uk_ui.rs:605]: ignore walley for now

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 4]
```diff
@@ -197,18 +235,284 @@ async fn should_make_adyen_clearpay_payment(driver: WebDriver) -> Result<(), Web
...
+    let conn = AdyenSeleniumTest {};
+    conn.make_paypal_payment(
+        web_driver,
+        &format!("{CHEKOUT_BASE_URL}/saved/202"),
         vec![
             Event::Assert(Assert::IsPresent("Google")),
             Event::Assert(Assert::ContainsAny(
                 Selector::QueryParamStr,
-                vec!["status=succeeded"],
+                vec!["status=processing"],
```
Comment (ArjunKarthik) [inline crates/router/tests/connectors/adyen_uk_ui.rs:253]: Why we are checking processing status instead of succeeded?

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]
For the payment methods which have processing as their final status lets add comment

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 5]
```diff
@@ -197,32 +235,294 @@ async fn should_make_adyen_clearpay_payment(driver: WebDriver) -> Result<(), Web
...
+}
+
+async fn should_make_adyen_bacs_payment(web_driver: WebDriver) -> Result<(), WebDriverError> {
+    let conn = AdyenSeleniumTest {};
+    conn.make_redirection_payment(
+        web_driver,
+        vec![
+            Event::Trigger(Trigger::Goto(&format!("{CHEKOUT_BASE_URL}/saved/54"))),
+            Event::Trigger(Trigger::Click(By::Id("card-submit-btn"))),
+            Event::Assert(Assert::IsPresent("processing")), //final status of this payment method will remain in processing state
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen_uk_ui.rs:296]:
```suggestion
            Event::Assert(Assert::IsPresent("Status")),
            Event::Assert(Assert::IsPresent("processing")), //final status of this payment method will remain in processing state
```

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 6]
```diff
@@ -197,32 +235,294 @@ async fn should_make_adyen_clearpay_payment(driver: WebDriver) -> Result<(), Web
...
+}
+
+async fn should_make_adyen_blik_payment(web_driver: WebDriver) -> Result<(), WebDriverError> {
+    let conn = AdyenSeleniumTest {};
+    conn.make_redirection_payment(
+        web_driver,
+        vec![
+            Event::Trigger(Trigger::Goto(&format!("{CHEKOUT_BASE_URL}/saved/64"))),
+            Event::Trigger(Trigger::Click(By::Id("card-submit-btn"))),
+            Event::Assert(Assert::IsPresent("processing")), //final status of this payment method will remain in processing state
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/adyen_uk_ui.rs:340]:
```suggestion
            Event::Assert(Assert::IsPresent("Status")),
            Event::Assert(Assert::IsPresent("processing")), //final status of this payment method will remain in processing state
```

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by ashokkjag — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Pull request merged**